dpsctl/dpsctl.py

```diff
@@ -608,6 +646,362 @@ def create_comms(args):
+    a_dac_upper_range = calibration_a_dac[last_point] - (calibration_a_dac[last_point] - calibration_a_dac[first_point]) * 0.1
+
+    # Draw data in graph
+    if calibration_debug_plotting:
+        plt.title("Output Current Sweep")
+        plt.xlabel("A_DAC")
+        x_data = list(calibration_a_dac)
+        plt.ylabel("A_ADC")
+        y_data = list(calibration_a_adc)
+        plt.plot(x_data, y_data, 'ro')
+        plt.axvline(x=a_dac_lower_range)
+        plt.axvline(x=a_dac_upper_range)
+        plt.axis(xmin=0, ymin=0)
+        plt.show()
+
+    # Take multiple current readings in this range
+    num_steps = 15
+    calibration_i_out = []
+    calibration_a_dac = []
+
+    for x in range(num_steps):
+        # Calculate our output current DAC value
+        output_dac = int(a_dac_lower_range + ((a_dac_upper_range - a_dac_lower_range) * (x / num_steps)))
+
+        # Set the output current
+        args.parameter = ["A_DAC={}".format(output_dac)]
+        payload = create_set_parameter(args.parameter)
+        communicate(comms, payload, args)
+        communicate(comms, create_enable_output("on"), args)
+        time.sleep(1)  # Wait for the DPS output to settle
+
+        # Add these readings to our array
+        calibration_i_out.append((get_average_calibration_result(comms, 'vout_adc') * v_adc_k + v_dac_c) / load_resistance)
+        calibration_a_dac.append(output_dac)
+
+    communicate(comms, create_enable_output("off"), args)  # Turn the output off
+
+    # Calculate and set the A_DAC coeffecients
+    a_dac_k, a_dac_c = best_fit(calibration_i_out, calibration_a_dac)
+    args.calibration_set = ['A_DAC_K={}'.format(a_dac_k), 'A_DAC_C={}'.format(a_dac_c)]
+    payload = create_set_calibration(args.calibration_set)
+    communicate(comms, payload, args)
+    print("\r\nConstant Current Calibration Complete\r\n")
+
+    # Draw data in graph
+    if calibration_debug_plotting:
+        plt.title("Output Current Calibration (A_DAC)")
+        ax = plt.gca()
+        plt.text(0.03, 0.97, 'y = {} * x + {}'.format(a_dac_k, a_dac_c), transform=ax.transAxes, fontsize=9, va='top')
+        plt.xlabel("Output Current (A)")
+        x_data = list(calibration_i_out)
+        plt.ylabel("A_DAC")
+        y_data = list(calibration_a_dac)
+        plt.plot(x_data, y_data, 'ro')
+        y_data = []
+        x_data.insert(0, 0)  # Ensure we have a x = 0 point on our line of best fit
+        for i in range(len(x_data)):
+            y_data.append(x_data[i] * a_dac_k + a_dac_c)
+        plt.plot(x_data, y_data, '-')  # Draw line of best fit
+        plt.axis(xmin=0, ymin=0)
+        plt.show()
+
+    # Change to the main screen
+    communicate(comms, create_change_screen(protocol.CHANGE_SCREEN_MAIN), args)
+
+    print("Calibration Complete\r\n")
+    print("To restore the device to factory defaults use dpsctl.py --calibration_reset")
```

Comment:
A bit picky-ish perhaps but should the factory part be left out of the wording as it suggests it is possible to return to an original setting that was lost the the DPS was re-flashed?

Reply:
Yeah that's a good point. Perhaps it should be changed to opendps defaults?

Reply:
I've added this to #122